### PR TITLE
Retire quarkus-junit4

### PR DESCRIPTION
**Pull request by gastaldi**
This PR removes the `quarkus-junit4` and any references to the JUnit 4 JAR in the BOM. 

It also migrates the JUnit 4 tests to JUnit 5

Discussion in https://groups.google.com/forum/#!topic/quarkus-dev/2YFVhwYv2N4

### CODE REVIEW

File: bom/runtime/pom.xml [hunk 1]
```diff
@@ -1096,11 +1089,6 @@
                 <artifactId>javax.ws.rs-api</artifactId>
                 <version>${javax.ws.rs-api.version}</version>
             </dependency>
-            <dependency>
```
Comment (kenfinnigan) [inline bom/runtime/pom.xml]: Have you run this with TCKs enabled?

I suspect removing the junit dependency will break them

Comment (gastaldi) [reply bom/runtime/pom.xml]: I browsed the POMs, and noticed they depend on `quarkus-arquillian`, which transitively resolves JUnit 4 through `arquillian-junit-container`, but I'll double check

Comment (kenfinnigan) [reply bom/runtime/pom.xml]: The `common` module has a dependency on junit.

It might come in through Arq BOM

Comment (gastaldi) [reply bom/runtime/pom.xml]: Not only that, but the tests in other modules (including the core) depend on JUnit4. I'll make sure to migrate them as well

**Comment by gastaldi** [comment]
The PR is now ready to be reviewed.

**Review by stuartwdouglas — APPROVED** [review]

**Comment by stuartwdouglas** [comment]
This LGTM, not sure if @kenfinnigan or @gsmet also want to look it over?

**Review by gsmet — APPROVED** [review]

**Comment by rsvoboda** [comment]
@gsmet should we have `noteworthy-removal` label to track things like this for release notes ?

**Pull request merged**